### PR TITLE
Add volumegroup sidecar to CSI driver operator

### PR DESCRIPTION
**Pull request by ELENAGER**
Signed-off-by: Elena Gershkovich <elenage@il.ibm.com>

### CODE REVIEW

File: controllers/syncer/csi_controller.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ const (
 	snapshotterContainerName             = "csi-snapshotter"
```
Comment (matancarmeli7) [inline controllers/syncer/csi_controller.go:46]: can you add `volumeGroupContainerName` to [this case](https://github.com/IBM/ibm-block-csi-operator/blob/166783ed3bab275049decb4eb4f1d15681c729d3/controllers/syncer/csi_controller.go#L336)?

**Review by matancarmeli7 — COMMENTED** [review]
Can you update the example CR and the roles in the CSV files of our project? like we did in the replication [PR](https://github.com/IBM/ibm-block-csi-operator/pull/209/files#diff-3c3edfd886e1edeb9de30dc4d15b557c0a0c3d19a049a9a0e3fa3b7b3b045e0e)
those are the files:

- [community](https://github.com/IBM/ibm-block-csi-operator/blob/develop/deploy/olm-catalog/ibm-block-csi-operator-community/1.11.0/manifests/ibm-block-csi-operator.v1.11.0.clusterserviceversion.yaml)
- [certified](https://github.com/IBM/ibm-block-csi-operator/blob/develop/deploy/olm-catalog/ibm-block-csi-operator/1.11.0/manifests/ibm-block-csi-operator.clusterserviceversion.yaml)

File: controllers/syncer/csi_controller.go (hunk 1 above)
Comment (ELENAGER) [reply controllers/syncer/csi_controller.go:46]: Done

File: controllers/internal/crutils/static_resource_generator.go [hunk 2]
```diff
@@ -397,6 +404,76 @@ func (c *IBMBlockCSI) GenerateCSIAddonsReplicatorClusterRoleBinding() *rbacv1.Cl
...
+				Verbs:     []string{verbUpdate},
+			},
+			{
+				APIGroups: []string{volumeGroupApiGroup},
+				Resources: []string{volumeGroupClassesResource},
+				Verbs:     []string{verbGet, verbList, verbWatch},
+			},
+			{
+				APIGroups: []string{volumeGroupApiGroup},
+				Resources: []string{volumeCroupContentsResource},
```
Comment (ArbelNathan) [inline controllers/internal/crutils/static_resource_generator.go:435]: volumeGroupContentsResource (not Croup)

File: controllers/internal/crutils/static_resource_generator.go [hunk 3]
```diff
@@ -397,6 +404,76 @@ func (c *IBMBlockCSI) GenerateCSIAddonsReplicatorClusterRoleBinding() *rbacv1.Cl
...
+			},
+			{
+				APIGroups: []string{volumeGroupApiGroup},
+				Resources: []string{volumeGroupClassesResource},
+				Verbs:     []string{verbGet, verbList, verbWatch},
+			},
+			{
+				APIGroups: []string{volumeGroupApiGroup},
+				Resources: []string{volumeGroupContentsResource},
+				Verbs:     []string{verbGet, verbList, verbWatch},
```
Comment (matancarmeli7) [inline controllers/internal/crutils/static_resource_generator.go:436]: can you add to `volumeGroupContentsResource` the `Create` verb? (I know it's not written in the sidecar I will add it this week)?

File: controllers/internal/crutils/static_resource_generator.go (hunk 2 above)
Comment (ELENAGER) [reply controllers/internal/crutils/static_resource_generator.go]: Thanks, done

**Review by matancarmeli7 — APPROVED** [review]
LGTM

**Review by ArbelNathan — APPROVED** [review]
LGTM

**Pull request merged**